.github/workflows/publish.yml

```diff
@@ -0,0 +1,39 @@
+name: publish
+
+on:
+  push:
+    paths:
+      - "src/**"
+      - ".github/workflows/publish.yml"
+    branches:
+      - main
+  pull_request:
+    paths:
+      - "src/**"
+      - ".github/workflows/publish.yml"
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        machine: [windows-latest, macos-latest, ubuntu-20.04]
+        architecture: [x64, arm64]
+        include:
+          - machine: windows-latest
+            platform: win
+          - machine: macos-latest
+            platform: osx
+          - machine: ubuntu-20.04
+            platform: linux
+    runs-on: ${{ matrix.machine }}
+    timeout-minutes: 5
+    steps:
+      - name: checkout
+        uses: actions/checkout@v1
+      - name: install linux prerequisites
+        if: matrix.platform == "linux"
+        run: |
+          sudo apt-get install clang zlib1g-dev
+          sudo apk add clang build-base zlib-dev
```

Comment:
you're mixing instructions for two different distros here. only the first line applies for Ubuntu